tests/test_unittest/test_merge_blocks_pass.py

```diff
@@ -106,6 +106,23 @@ def test_merge_write_after_read_k_parallel(
     assert len(transform_data.blocks) == 2
 
 
+def test_no_merge_read_with_offset_after_write(
+    merge_blocks_pass: PassType, ijk_domain: Domain
+) -> None:
+    transform_data = make_transform_data_multiple(
+        name="no_merge_k_offset",
+        domain=ijk_domain,
+        fields=["out", "in", "tmp"],
+        info=[
+            ([("tmp", "in", (0, 0, 1))], IterationOrder.FORWARD, (1, -1)),
+            ([("out", "tmp", (0, 0, -1))], IterationOrder.FORWARD, (1, -2)),
+        ],
+    )
+    transform_data = merge_blocks_pass(transform_data)
+    # not allowed to be merged into the same stage: first block should have 2 IJ blocks
+    assert len(transform_data.blocks[0].ij_blocks) == 2
+
```

Comment:
Ideally we should also check that the two multistages are kept in the right order. Especially since a reordering pass is planned.

Reply:
Good idea. I'll make that change.